include/mbedtls/ecp.h

```diff
@@ -501,6 +501,50 @@ int mbedtls_ecp_tls_read_group( mbedtls_ecp_group *grp, const unsigned char **bu
 int mbedtls_ecp_tls_write_group( const mbedtls_ecp_group *grp, size_t *olen,
                          unsigned char *buf, size_t blen );
 
+#if defined(MBEDTLS_ASN1_WRITE_C) && defined(MBEDTLS_OID_C)
+/**
+ * \brief           Maximum size of the output of mbedtls_ecp_ansi_write_group
+ */
+#define MBEDTLS_OID_EC_GRP_MAX_SIZE 12
+
+/**
+ * \brief           Write the ANSI X9.62/RFC5480 OID ECParameters of a group
+ *
+ *                  The output is the group's OID wrapped as ASN.1.
+ *
+ * \param grp       ECP group used
+ * \param buf       Buffer to write to
+ * \param size      Buffer size
+ *
+ * \return          Number of bytes written to \c buf,
+ *                  or \c MBEDTLS_ERR_ECP_BUFFER_TOO_SMALL
+ *                  or \c MBEDTLS_ERR_OID_NOT_FOUND
+ */
+int mbedtls_ecp_ansi_write_group( const mbedtls_ecp_group *grp,
```

Comment:
I would prefer returning the actually used buffer size through a pointer argument, as we're doing in other places, too.

Reply:
Indeed. @AndrzejKurek When I added the extra auxiliary functions to support ECDSA in pkcs11_client, I initially wrote the code inline in pkcs11_client.c and moved the code to another module with a minimum of changes. There may be other similar things to improve to make this a proper function on its own.

Reply:
Done